test/cpp/tensorexpr/test_cuda.cpp

```diff
@@ -2344,6 +2344,315 @@ TEST(Cuda, MaskMultiDimMultiLevel_CUDA) {
   cudaFree(d_dev);
 }
 
+TEST(Cuda, DependencyAcrossLoops_CUDA) {
+  // for (int x = 0; x < 10; x++) {
+  //   b[x] = a[x];
+  // }
+  // for (int y = 0; y < 10; y++) {
+  //   c[y] = b[y+1];
+  // }
+  KernelScope kernel_scope;
+
+  Placeholder a("a", kInt, {50});
+  Tensor* b =
+      Compute("b", {{50, "x"}}, [&](const VarHandle& i) { return a.load(i); });
+  Tensor* c = Compute(
+      "c", {{50, "y"}}, [&](const VarHandle& i) { return b->call(i + 1); });
+  LoopNest l({b, c});
+  std::vector<For*> loops = l.getLoopStmtsFor(b);
+  l.setGPUThreadIndex(loops[0], 0);
+  loops = l.getLoopStmtsFor(c);
+  l.setGPUThreadIndex(loops[0], 0);
+
+  l.prepareForCodegen();
+  Stmt* s = l.root_stmt();
+
+  CudaCodeGen cg(s, {a, b, c});
+  std::ostringstream oss;
+  oss << *cg.stmt();
+
+  const std::string& verification_pattern =
+      R"IR(
+        # CHECK: b[threadIdx.x] =
+        # CHECK: __syncthreads();
```

Comment:
We need to check that the __syncthreads is outside both loops.

Reply:
This code is after the loops have been removed, i.e., after `CudaCodeGen`. There are no loops at that point in this example. Am I missing something here?